### PR TITLE
add move

### PR DESCRIPTION
**Pull request by luanshd**
Signed-off-by: Luan Shengde <shdluan@163.com>

### CODE REVIEW

**Comment by caspiano** [comment]
Thanks for the addition, any chance you could add a spec?

File: src/etcd/kv.cr [hunk 1]
```diff
@@ -126,6 +126,40 @@ module Etcd
       Model::TxnResponse.from_json(response.body).succeeded
     end
 
+    def move(key : String,  key_d : String, value, lease : Int64 = 0_i64) : Bool
+      key_o = Base64.strict_encode(key)
+      key_d = Base64.strict_encode(key_d)
```
Comment (caspiano) [inline src/etcd/kv.cr:131]:
```suggestion
    def move(key : String,  key_destination : String, value, lease : Int64 = 0_i64) : Bool
      key_o = Base64.strict_encode(key)
      key_d = Base64.strict_encode(key_destination)
```

File: src/etcd/kv.cr [hunk 2]
```diff
@@ -126,6 +126,40 @@ module Etcd
       Model::TxnResponse.from_json(response.body).succeeded
     end
 
+    def move(key : String,  key_d : String, value, lease : Int64 = 0_i64) : Bool
```
Comment (caspiano) [inline src/etcd/kv.cr:129]: As this is a constructed transaction, could you describe the operation?

Comment (luanshd) [reply src/etcd/kv.cr:129]: sometime, we may have different class to store data, something like queue,

Comment (luanshd) [reply src/etcd/kv.cr:129]: sometime we may store data with different key prefix, some thing like different queues.

Comment (luanshd) [reply src/etcd/kv.cr:129]: Sometime we may store data with different key prefix, some thing like different queues. When we handle the data, we have demand to add  a new key with the same value, and at the same time, we need to delete the old key-value. We need to do them at the same time. so add function move to resolve it.
thanks.

File: src/etcd/kv.cr (hunk 1 above)
Comment (luanshd) [reply src/etcd/kv.cr:131]: It's OK.
Thanks

File: src/etcd/kv.cr [hunk 3]
```diff
@@ -126,6 +126,40 @@ module Etcd
       Model::TxnResponse.from_json(response.body).succeeded
     end
 
+    def move(key : String,  key_destination : String, value, lease : Int64 = 0_i64) : Bool
```
Comment (caspiano) [inline src/etcd/kv.cr:129]:
```suggestion
    # Moves a value from `key` to `key_destination`, deleting the kv at `key` in the process.
    def move(key : String,  key_destination : String, value, lease : Int64 = 0_i64) : Bool
```

**Pull request merged**